File: coffee_machine_java/src/IngredientManager.java
// src/IngredientManager.java
public class IngredientManager {
    private int water;
    private int beans;
    private int milk;

    public IngredientManager() {
        this.water = 1000;  // 1000 ml vett
        this.beans = 500;   // 500 g ube
        this.milk = 500;    // 500 ml piima
    }

    public boolean hasIngredients(Beverage beverage) {
        return water >= beverage.getWater() && beans >= beverage.getBeans() && milk >= beverage.getMilk();
    }

    public void useIngredients(Beverage beverage) {
        water -= beverage.getWater();
        beans -= beverage.getBeans();
        milk -= beverage.getMilk();
    }

    public void refillIngredients(int water, int beans, int milk) {
        this.water += water;
        this.beans += beans;
        this.milk += milk;
    }
}

File: coffee_machine_python/ingredient_manager.py
# ingredient_manager.py
class IngredientManager:
    def __init__(self):
        self.water = 1000  # ml
        self.beans = 500  # grams
        self.milk = 500  # ml

    def has_ingredients(self, coffee):
        return (self.water >= coffee.water and
                self.beans >= coffee.beans and
                self.milk >= coffee.milk)

    def use_ingredients(self, coffee):
        self.water -= coffee.water
        self.beans -= coffee.beans
        self.milk -= coffee.milk

    def refill_ingredients(self, water=0, beans=0, milk=0):
        self.water += water
        self.beans += beans
        self.milk += milk
